source/includes/databases-collections/databases-collections.scala

```diff
@@ -0,0 +1,101 @@
+import org.mongodb.scala._
+import org.mongodb.scala.bson.{Document, ObjectId}
+import org.mongodb.scala.gridfs.GridFSBucket
+import org.mongodb.scala.model._
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.TimeUnit
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.jdk.CollectionConverters.SeqHasAsJava
+
+object DatabasesCollections {
+
+  def main(args: Array[String]): Unit = {
+    val mongoClient = MongoClient("<connection string URI>")
+
+    // start-access-database
+    val database = mongoClient.getDatabase("test_database")
+    // end-access-database
+
+    {
+      // start-access-collection
+      val collection = database.getCollection("test_collection")
+      // end-access-collection
+    }
+
+    // start-create-collection
+    val createObservable = database.createCollection("example_collection")
+    Await.result(createObservable.toFuture(), Duration(10, TimeUnit.SECONDS))
+    // end-create-collection
+
+    {
+      // start-find-collections
+      val results = Await.result(database.listCollections().toFuture(), Duration(10, TimeUnit.SECONDS))
+      results.foreach(println)
+      // end-find-collections
+    }
+
+    {
+      // start-find-collection-names
+      val results = Await.result(database.listCollectionNames().toFuture(), Duration(10, TimeUnit.SECONDS))
+      results.foreach(println)
+      // end-find-collection-names
+    }
+
+    {
+      // start-delete-collection
+      val deleteObservable = database.getCollection("test_collection").drop()
+      Await.result(deleteObservable.toFuture(), Duration(10, TimeUnit.SECONDS))
+      // end-delete-collection
+    }
+
+    // start-database-read-prefs
+    val databaseWithReadPrefs = mongoClient.getDatabase("test_database").withReadPreference(ReadPreference.secondary())
+    // end-database-read-prefs
```

Comment:
[s] When reading this in the snippet, this line looks a bit long and requires some horizontal scrolling. Maybe the spacing can be reformatted for readability?